onnxruntime/test/providers/kernel_def_hash_test.cc

```diff
@@ -63,8 +63,10 @@
 #pragma warning(disable : 28020)
 #elif __aarch64__
 #pragma GCC diagnostic push
+#if !defined(__has_warning) || __has_warning("-Wformat-truncation")
```

Comment:
if __has_warning is not defined, is the rest of the line valid?

Reply:
Yes, __has_warning  seems to be only defined for clang. gcc doesn't have it, as we need to ignore warning if using gcc

Reply:
In https://github.com/microsoft/onnxruntime/blob/master/cmake/CMakeLists.txt#L917
Add  
```
 check_cxx_compiler_flag(-Wformat-truncation HAS_FORMAT_TRUNCATION)
```
Then open https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime_config.h.in, add HAS_FORMAT_TRUNCATION there.

Then in this file, 
```c++
#include <onnxruntime_config.h>

#ifdef HAS_FORMAT_TRUNCATION 
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wformat-truncation"
#endif
```

Reply:
Yep, I think this is simpler